### PR TITLE
 adds a reader for compliance data types. moves the list compliance da

### PR DESCRIPTION
**Pull request by theseyi**
tatypes type definitions to the list directory.

 - implements getFildIdentifierOptions to extract dropdown options from complianceDataTypes list returned from GET /complianceDataTypes endpoint. adds unit tests. adds mirage mock for complianceDataTypes response and fixtures.

### CODE REVIEW

File: wherehows-web/app/typings/api/list/compliance-datatypes.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Describes the interface for a complianceDataType
+ * @export
+ * @interface IComplianceDataType
+ */
+export interface IComplianceDataType {
+  // Indicates that id contains pii
+  pii: boolean;
+  // Flag indicating if a field with this id is a member identifier
+  idType: boolean;
```
Comment (cptran777) [inline wherehows-web/app/typings/api/list/compliance-datatypes.d.ts]: idType being a boolean strikes me as a weird way to name the property.

File: wherehows-web/app/utils/api/list/compliance-datatypes.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+const readComplianceDataTypes = async (): Promise<Array<IComplianceDataType>> => {
+  const { status, complianceDataTypes = [], msg } = await getJSON<IComplianceDataTypeResponse>({
+    url: complianceDataTypesUrl
+  });
+
+  if (status === ApiStatus.OK && complianceDataTypes.length) {
+    return complianceDataTypes;
+  }
+
+  throw new Error(msg);
```
Comment (cptran777) [inline wherehows-web/app/utils/api/list/compliance-datatypes.ts]: I'm assuming there's a catch somewhere to handle this thrown error?

File: wherehows-web/app/typings/api/list/compliance-datatypes.d.ts (hunk 1 above)
Comment (theseyi) [reply wherehows-web/app/typings/api/list/compliance-datatypes.d.ts]: Yea, I feel the same way too, but unfortunately, this is a mirror for the api response shape :/ @cptran777

File: wherehows-web/app/utils/api/list/compliance-datatypes.ts (hunk 2 above)
Comment (theseyi) [reply wherehows-web/app/utils/api/list/compliance-datatypes.ts]: Yea, that's the intention, the consumer will have to implement error handling

**Pull request merged**